main/dzVents.cpp

```diff
@@ -662,14 +662,14 @@ void CdzVents::IterateTable(lua_State *lua_state, const int tIndex, std::vector<
 		}
 		else if (std::string(luaL_typename(lua_state, -1)) == "number")
 		{
-			item.type = TYPE_INTEGER;
-			item.iValue = (int)lua_tointeger(lua_state, -1);
+			item.type = TYPE_FLOAT;
+			item.fValue = (float)lua_tonumber(lua_state, -1);
 			item.name = std::string(lua_tostring(lua_state, -2));
 		}
 		else if (std::string(luaL_typename(lua_state, -1)) == "boolean")
 		{
 			item.type = TYPE_BOOLEAN;
-			item.iValue = lua_toboolean(lua_state, -1);
+			item.fValue = lua_toboolean(lua_state, -1);
```

Comment:
I would not put a boolean value in a float. That will result in failing boolean checks. Currently it does not harm because the value is not used in case of a boolean, but this would fail if the value is used. Keep iValue here instead of fValue.